### ISSUE TITLE
Diagnostics without autofix are not reported when `--fix` flag is provided

### ISSUE THREAD

**Issue by Cerber-Ursi** [Antonboom/testifylint#251]
**Reproduction steps:**

- Get a file that violates some non-autofixable diagnostic, e.g.:
    ```golang
    package requireerror

    import (
        	"github.com/stretchr/testify/assert"
    )

    func TestRequireErrorChecker(t *testing.T) {
    	assert.Error(t, err)  
    }
    ```
- Run `testifylint --fix ./...`.

**Expected:** linter emits the same diagnostic as it would do without `--fix`.
**Actual:** linter finishes without emitting anything.

Not sure if it is testifylint's fault or something common for Go linters in general, to be honest, so feel free to redirect me if this should be discussed somewhere else.

**Comment by Antonboom** [comment]
Compilable and reproducible example:
```go
package requireerror

import (
	"testing"
	"github.com/stretchr/testify/assert"
)

func TestRequireErrorChecker(t *testing.T) {
	var err error
	assert.Error(t, err)
	foo()
}

func foo() {}
```

```
$ testifylint test_test.go
/tmp/test/test_test.go:10:2: require-error: for error assertions use require

$ testifylint -fix test_test.go
```
